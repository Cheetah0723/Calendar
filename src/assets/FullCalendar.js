import 'rc-calendar/assets/index.less';
import 'rc-select/assets/index.css';
