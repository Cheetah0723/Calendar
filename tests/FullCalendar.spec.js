/* eslint-disable no-undef */
import React from 'react';
import Select from 'rc-select';
import { render } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import moment from 'moment';
import FullCalendar from '../src/FullCalendar';
import zhCN from '../src/locale/zh_CN';
import enUS from '../src/locale/en_US';

describe('FullCalendar', () => {
  it('render correctly', () => {
    const zhWrapper = render(
      <FullCalendar
        type="month"
        Select={Select}
        locale={zhCN}
        defaultValue={moment().locale('zh-cn')}
      />
    );
    expect(renderToJson(zhWrapper)).toMatchSnapshot();

    const enWrapper = render(
      <FullCalendar
        type="month"
        Select={Select}
        locale={enUS}
        defaultValue={moment().locale('en')}
      />
    );
    expect(renderToJson(enWrapper)).toMatchSnapshot();
  });
});
