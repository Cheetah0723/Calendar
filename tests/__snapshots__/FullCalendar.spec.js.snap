// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullCalendar render correctly 1`] = `
<div
  class="rc-calendar rc-calendar-full"
  tabindex="0"
>
  <div
    class="rc-calendar-full-header"
  >
    <span
      class="rc-calendar-full-header-switcher"
    >
      <span
        class="rc-calendar-full-header-switcher-normal"
      >
        月
      </span>
      <span
        class="rc-calendar-full-header-switcher-focus"
      >
        年
      </span>
    </span>
    <div
      class="rc-calendar-full-header-year-select rc-select rc-select-enabled"
    >
      <div
        aria-autocomplete="list"
        aria-expanded="false"
        aria-haspopup="true"
        class="rc-select-selection
                  rc-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="rc-select-selection__rendered"
        >
          <div
            class="rc-select-selection-selected-value"
            style="display:block;opacity:1;"
            title="2017"
          >
            2017
          </div>
        </div>
        <span
          class="rc-select-arrow"
          style="user-select:none;-webkit-user-select:none;"
          unselectable="unselectable"
        >
          <b />
        </span>
      </div>
    </div>
  </div>
  <div
    class="rc-calendar-calendar-body"
  >
    <table
      cellspacing="0"
      class="rc-calendar-month-panel-table"
      role="grid"
    >
      <tbody
        class="rc-calendar-month-panel-tbody"
      >
        <tr
          role="row"
        >
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="一月"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              一月
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="二月"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              二月
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell rc-calendar-month-panel-selected-cell rc-calendar-month-panel-current-cell"
            role="gridcell"
            title="三月"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              三月
            </a>
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="四月"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              四月
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="五月"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              五月
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="六月"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              六月
            </a>
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="七月"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              七月
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="八月"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              八月
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="九月"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              九月
            </a>
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="十月"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              十月
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="十一月"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              十一月
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="十二月"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              十二月
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`FullCalendar render correctly 2`] = `
<div
  class="rc-calendar rc-calendar-full"
  tabindex="0"
>
  <div
    class="rc-calendar-full-header"
  >
    <span
      class="rc-calendar-full-header-switcher"
    >
      <span
        class="rc-calendar-full-header-switcher-normal"
      >
        Month
      </span>
      <span
        class="rc-calendar-full-header-switcher-focus"
      >
        Year
      </span>
    </span>
    <div
      class="rc-calendar-full-header-year-select rc-select rc-select-enabled"
    >
      <div
        aria-autocomplete="list"
        aria-expanded="false"
        aria-haspopup="true"
        class="rc-select-selection
                  rc-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="rc-select-selection__rendered"
        >
          <div
            class="rc-select-selection-selected-value"
            style="display:block;opacity:1;"
            title="2017"
          >
            2017
          </div>
        </div>
        <span
          class="rc-select-arrow"
          style="user-select:none;-webkit-user-select:none;"
          unselectable="unselectable"
        >
          <b />
        </span>
      </div>
    </div>
  </div>
  <div
    class="rc-calendar-calendar-body"
  >
    <table
      cellspacing="0"
      class="rc-calendar-month-panel-table"
      role="grid"
    >
      <tbody
        class="rc-calendar-month-panel-tbody"
      >
        <tr
          role="row"
        >
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="Jan"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              Jan
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="Feb"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              Feb
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell rc-calendar-month-panel-selected-cell rc-calendar-month-panel-current-cell"
            role="gridcell"
            title="Mar"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              Mar
            </a>
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="Apr"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              Apr
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="May"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              May
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="Jun"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              Jun
            </a>
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="Jul"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              Jul
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="Aug"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              Aug
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="Sep"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              Sep
            </a>
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="Oct"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              Oct
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="Nov"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              Nov
            </a>
          </td>
          <td
            class="rc-calendar-month-panel-cell"
            role="gridcell"
            title="Dec"
          >
            <a
              class="rc-calendar-month-panel-month"
            >
              Dec
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
